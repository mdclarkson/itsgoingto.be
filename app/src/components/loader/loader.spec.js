import React from 'react'
import { render } from '@testing-library/react'
import Loader from './loader'

describe('(Component) Loader', () => {
  describe('(Render)', () => {
    describe('when prop isLoading is false', () => {
      it('matches snapshot', () => {
        const { asFragment } = render(<Loader isLoading={false} />)

        expect(asFragment()).toMatchSnapshot()
      })
    })

    describe('when prop isLoading is true', () => {
      it('matches snapshot', () => {
        const { asFragment } = render(<Loader isLoading />)

        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
})
