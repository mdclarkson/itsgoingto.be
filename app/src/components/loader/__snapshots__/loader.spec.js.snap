// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Loader (Render) when prop isLoading is false matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="loader-container hideable gone"
  >
    <div
      class="loader"
    >
      <p
        class="loader__label"
      >
        ?
      </p>
      <div
        class="loader__figure"
      />
    </div>
  </div>
</DocumentFragment>
`;

exports[`(Component) Loader (Render) when prop isLoading is true matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="loader-container hideable"
  >
    <div
      class="loader"
    >
      <p
        class="loader__label"
      >
        ?
      </p>
      <div
        class="loader__figure"
      />
    </div>
  </div>
</DocumentFragment>
`;
